src/Domain/CoordinateMaps/FocallyLiftedEndcap.hpp

```diff
@@ -0,0 +1,426 @@
+ * \left(\bar{\rho} \frac{dq}{d\bar{\rho}} + q\right)^{-1},\\
+ * \frac{\partial q}{\partial x_0^1} &=
+ * \frac{\bar{y}}{\bar{\rho}R}\frac{dq}{d\bar{\rho}}
+ * \left(\bar{\rho} \frac{dq}{d\bar{\rho}} + q\right)^{-1},
+ * \f}
+ *
+ * where we have differentiated Eq. (13), and where we have
+ * used Eqs. (15) and (16) to eliminate \f$x_0^0\f$ and
+ * \f$x_0^1\f$ in favor of \f$\bar{x}\f$ and
+ * \f$\bar{y}\f$ in the final result.
+ *
+ * By differentiating Eqs. (15) and (16), and using Eqs. (17) and (18), we
+ * find
+ *
+ * \f{align*}
+ * \frac{\partial \bar{x}}{\partial x_0^0} &=
+ * \frac{1}{R q}
+ * - \frac{\bar{x}^2}{R q \bar{\rho}} \frac{dq}{d\bar{\rho}}
+ * \left(\bar{\rho} \frac{dq}{d\bar{\rho}} + q\right)^{-1},\\
+ * \frac{\partial \bar{x}}{\partial x_0^1} &=
+ * - \frac{\bar{x}\bar{y}}{R q \bar{\rho}} \frac{dq}{d\bar{\rho}}
+ * \left(\bar{\rho} \frac{dq}{d\bar{\rho}} + q\right)^{-1},\\
+ * \frac{\partial \bar{x}}{\partial x_0^2} &= 0,\\
+ * \frac{\partial \bar{y}}{\partial x_0^0} &=
+ * \frac{\partial \bar{x}}{\partial x_0^1},\\
+ * \frac{\partial \bar{y}}{\partial x_0^1} &=
+ * \frac{1}{R q}
+ * - \frac{\bar{y}^2}{R q \bar{\rho}} \frac{dq}{d\bar{\rho}}
+ * \left(\bar{\rho} \frac{dq}{d\bar{\rho}} + q\right)^{-1},\\
+ * \frac{\partial \bar{y}}{\partial x_0^2} &= 0,\\
+ * \frac{\partial \bar{z}}{\partial x_0^i} &= 0.
+ * \f}
+ * Note that care must be taken to evaluate
+ * \f$q = \sin(\bar{\rho}\theta_\mathrm{max})/\bar{\rho}\f$ and its
+ * derivative near \f$\bar{\rho}=0\f$; see the discussion above on
+ * evaluating sinc functions.
+ *
+ * ### dxbar_dsigma
+ *
+ * `dxbar_dsigma` returns \f$\partial \bar{x}^i/\partial \sigma\f$,
+ *  where \f$x_0^i\f$ is held fixed.
+ *
+ * From Eq. (6) we have
+ *
+ * \f{align}
+ * \frac{\partial \bar{x}^i}{\partial \sigma} &= (0,0,2).
+ * \f}
+ *
+ */
+class Endcap {
+ public:
+  Endcap(const std::array<double, 3>& center, double radius,
+         double z_plane) noexcept;
+
+  Endcap() = default;
+  ~Endcap() = default;
+  Endcap(Endcap&&) = default;
+  Endcap(const Endcap&) = default;
+  Endcap& operator=(const Endcap&) = default;
+  Endcap& operator=(Endcap&&) = default;
+
+  template <typename T>
+  std::array<tt::remove_cvref_wrap_t<T>, 3> operator()(
+      const std::array<T, 3>& source_coords) const noexcept;
+
+  boost::optional<std::array<double, 3>> inverse(
+      const std::array<double, 3>& target_coords, double sigma_in) const
+      noexcept;
+
+  template <typename T>
+  tnsr::Ij<tt::remove_cvref_wrap_t<T>, 3, Frame::NoFrame> jacobian(
+      const std::array<T, 3>& source_coords) const noexcept;
+
+  template <typename T>
+  tnsr::Ij<tt::remove_cvref_wrap_t<T>, 3, Frame::NoFrame> inv_jacobian(
+      const std::array<T, 3>& source_coords) const noexcept;
+
+  template <typename T>
+  tt::remove_cvref_wrap_t<T> sigma(const std::array<T, 3>& source_coords) const
+      noexcept;
+
+  template <typename T>
+  std::array<tt::remove_cvref_wrap_t<T>, 3> deriv_sigma(
```

Comment:
Maybe I missed it, but it would be good to document why none of the `sigma`, `deriv_sigma`, etc. functions are used anywhere in this class